spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/QueueAttributesResolver.java

```diff
@@ -91,8 +92,18 @@ private CompletableFuture<String> resolveQueueUrl() {
 	}
 
 	private CompletableFuture<String> doResolveQueueUrl() {
+		return isValidQueueArn(this.queueName)
+			? doResolveQueueUrlFromArn()
+			: CompletableFutures.exceptionallyCompose(this.sqsAsyncClient.getQueueUrl(req -> req.queueName(this.queueName))
+				.thenApply(GetQueueUrlResponse::queueUrl), this::handleException);
+	}
+
+	private CompletableFuture<String> doResolveQueueUrlFromArn() {
```

Comment:
We can pass Arn here and skip the second call. 
`private CompletableFuture<String> doResolveQueueUrlFromArn(Arn arn) {`

Instead of doing second time `Arn.fromString`
`Arn arn = Arn.fromString(this.queueName);`

Reply:
Thanks for the suggestion, I have changed that as well.